### PR TITLE
Fix for integer overflow in contiguous-split

### PR DESCRIPTION
**Pull request by jbrennan333**
This is a fix for an integer overflow in `contiguous_split.cu::copy_data`. 

We encountered this problem when running a large query with the spark-rapids plug-in.  The plug-in OOM callback was reporting negative numbers for some failed allocations.  It treats the allocation size as a `long`.   The allocations were coming from `contiguous_split` with a large number of rows (399,000,000) of type `UINT64`.

With help from @nvdbaranec, I was able to isolate the problem to this line:
```
size_t const bytes = buf.num_elements * buf.element_size;
```

The two operands are both `INT32`.  So the result of the multiply is being sign-extended when assigning to `bytes`.

The fix is to cast these to `size_t` before the multiply.

### CODE REVIEW

File: cpp/src/copying/contiguous_split.cu
Comment (PointKernel) [inline cpp/src/copying/contiguous_split.cu:825]:
```suggestion
      auto const bytes =
        static_cast<std::size_t>(buf.num_elements) * static_cast<std::size_t>(buf.element_size);
```
Could you include `<cstddef>` and replace all `::size_t` with `std::size_t` throughout this function? The rest of the file is pretty consistent of using `std::size_t` instead of `::size_t`.

**Review by PointKernel — CHANGES_REQUESTED** [review]

File: cpp/src/copying/contiguous_split.cu
Comment (jbrennan333) [reply cpp/src/copying/contiguous_split.cu:825]: Thanks for the review!  Yes I will make this change.

**Review by PointKernel — APPROVED** [review]
Looks great. Thanks for helping clean up the code!

**Comment by PointKernel** [comment]
rerun tests

**Comment by jbrennan333** [comment]
They python build failure looks like build system issue:
```
[](https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-cpu-python-build/7219/console#L223)[](https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-cpu-python-build/7219/console#L224)
15:23:50 ‘EC2 (aws-b) - runner-m5d2xl (i-0fcffb338c35da323)’ doesn’t have label ‘cpu4’
15:23:50 ‘Jenkins’ doesn’t have label ‘cpu4’
15:49:46 [rapidsai » gpuci » cudf » prb » cudf-cpu-python-build » 11.5,3.9](https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-cpu-python-build/CUDA=11.5,PYTHON=3.9/)[](https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-cpu-python-build/7219/console#L225) completed with result FAILURE
15:49:46 [rapidsai » gpuci » cudf » prb » cudf-cpu-python-build » 11.5,3.8](https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci/job/cudf/job/prb/job/cudf-cpu-python-build/CUDA=11.5,PYTHON=3.8/) completed with result FAILURE
```

**Review by ttnghia — APPROVED** [review]

**Comment by jbrennan333** [comment]
Thanks for the review @ttnghia!

The build failures appear to be unrelated - do we need to wait until they are resolved before we can merge this?

**Comment by PointKernel** [comment]
The CI failure fix has been merged (see https://github.com/rapidsai/cudf/pull/10442). @jbrennan333 Could you merge with the latest `branch-22.04`?

**Comment by PointKernel** [comment]
@gpucibot merge

**Pull request merged**
